[plugin-lighthouse] Introduce Lighthouse plugin (#4002)
vividus-framework/vividus: docs/modules/plugins/pages/plugin-lighthouse.adoc

= Lighthouse Plugin

> The plugin provides the ability to generate Lighthouse performance reports on the user experience of a page on both mobile and desktop devices, and provides suggestions on how that page may be improved.

Reviewer: need to add link to Lighthouse

:plugin-short-name: lighthouse
include::partial$plugin-installation.adoc[]

== Properties
[cols="2,1,2,3", options="header"]
|===
|Property Name
|Acceptable values
|Default
|Description

|`lighthouse.application-name`
|<string>
|
|The application name to be used in the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent[User-Agent] header.

|`lighthouse.access-token`
|<string>
|
|The OAuth access token. Its recommended to specify the token if you plan to use the plugin in an automated way and making multiple queries per second.

|===

== Steps

=== Perform Lighthouse scan

Performs Lighthouse scan on the specified page and verifies performance result metrics.

[source,gherkin]
----

> When I perform Lighthouse scan on $strategy for `$url` page:$rules

Reviewer: > Lighthouse scan on $strategy

it doesn't sound like English sentence, need to rename `$strategy` parameter

> When I perform Lighthouse scan on $strategy for `$url` page:$rules

Reviewer: > $url

need to be more specific, like: `$webPageUrl`

----

* `$strategy` - The analysis strategy to use, either `desktop` or `mobile`.
* `$url` - The page URL to perform scan on.

> * `$rules` - The metric verification rules:

Reviewer: ```suggestion
* `$meticsValidations` - The metrics validations:
```

** `$metric` - The Lighthouse metrics such as:
*** `First Contentful Paint`
*** `Largest Contentful Paint`
*** `Total Blocking Time`
*** `Cumulative Layout Shift`
*** `Speed Index`
** `$rule` - xref:parameters:comparison-rule.adoc[The comparison rule].

> ** `$value` - The expected numeric metric value.

Reviewer: not clear what `numeric` is? is it integer/floating?

Author: numeric value refers to both integers and floating numbers

= Lighthouse Plugin

The plugin provides the ability to generate https://developer.chrome.com/docs/lighthouse/overview/[Lighthouse] performance reports on the user experience of a page on both mobile and desktop devices, and provides suggestions on how that page may be improved.

:plugin-short-name: lighthouse
include::partial$plugin-installation.adoc[]

== Properties
[cols="2,1,2,3", options="header"]
|===
|Property Name
|Acceptable values
|Default
|Description

|`lighthouse.application-name`
|<string>
|
|The application name to be used in the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent[User-Agent] header.

|`lighthouse.access-token`
|<string>
|

> |The OAuth access token. Its recommended to specify the token if you plan to use the plugin in an automated way and making multiple queries per second.

Reviewer: ```suggestion
|The OAuth access token. Its recommended to specify the token if you plan to use the plugin in an automated way and make multiple queries per second.
```

|===

== Steps

=== Perform Lighthouse scan

> Performs Lighthouse scan on the specified web page and verifies performance metrics results.

Reviewer: ```suggestion
Performs Lighthouse scan of the specified web page and validates performance metrics against expected thresholds.
```

[source,gherkin]
----
When I perform Lighthouse $scanType scan of `$webPageUrl` page:$meticsValidations
----

* `$scanType` - The scan type to use:
** `full` - The scan is performed on both `desktop` and `mobile` devices.
** `desktop` - The scan is performed on `desktop` device.
** `mobile` - The scan is performed on `mobile` device.
* `$webPageUrl` - The page URL to perform scan on.
* `$meticsValidations` - The metric validations:
** `$metric` - The Lighthouse metric name such as:
*** `First Contentful Paint`
*** `Largest Contentful Paint`
*** `Total Blocking Time`
*** `Cumulative Layout Shift`
*** `Speed Index`
** `$rule` - xref:parameters:comparison-rule.adoc[The comparison rule].

> ** `$value` - The expected integer or floating (e.g. `0.35`) number.

Reviewer: ```suggestion
** `$threshold` - The expected integer or floating (e.g. `0.35`) number.
```
